Build fails on Arch Linux

I have tried to build your app by executing

```
mkdir build
cd build
ccmake ..
```

Pressed `c` and got warning:

```
CMake Warning (dev) in CMakeLists.txt:
   No cmake_minimum_required command is present.  A line of code such as

     cmake_minimum_required(VERSION 3.13)

   should be added at the top of the file.  The version specified may be
 lower
   if you wish to support older CMake versions for this project.  For more
   information run "cmake --help-policy CMP0000".
 This warning is for project developers.  Use -Wno-dev to suppress it.
```

I had to press `e` at this moment, then `g` as requested. I ran `make` and got following errors:

```
/home/radek/slinker/src/MainFrame.cpp: In member function ‘bool MainFrame::AskUserForSolvingRulesFile()’:
/home/radek/slinker/src/MainFrame.cpp:288:15: error: ‘wxOPEN’ was not declared in this scope
   _T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
               ^~~~~~
/home/radek/slinker/src/MainFrame.cpp:288:15: note: suggested alternative: ‘wxFC_OPEN’
   _T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
               ^~~~~~
               wxFC_OPEN
/home/radek/slinker/src/MainFrame.cpp:288:22: error: ‘wxFILE_MUST_EXIST’ was not declared in this scope
   _T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
                      ^~~~~~~~~~~~~~~~~
/home/radek/slinker/src/MainFrame.cpp:288:22: note: suggested alternative: ‘wxFD_FILE_MUST_EXIST’
   _T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
                      ^~~~~~~~~~~~~~~~~
                      wxFD_FILE_MUST_EXIST
/home/radek/slinker/src/MainFrame.cpp: In member function ‘void MainFrame::OnConvertRulesFileToCCode(wxCommandEvent&)’:
/home/radek/slinker/src/MainFrame.cpp:589:15: error: ‘wxOPEN’ was not declared in this scope
   _T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
               ^~~~~~
/home/radek/slinker/src/MainFrame.cpp:589:15: note: suggested alternative: ‘wxFC_OPEN’
   _T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
               ^~~~~~
               wxFC_OPEN
/home/radek/slinker/src/MainFrame.cpp:589:22: error: ‘wxFILE_MUST_EXIST’ was not declared in this scope
   _T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
                      ^~~~~~~~~~~~~~~~~
/home/radek/slinker/src/MainFrame.cpp:589:22: note: suggested alternative: ‘wxFD_FILE_MUST_EXIST’
   _T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
                      ^~~~~~~~~~~~~~~~~
                      wxFD_FILE_MUST_EXIST
/home/radek/slinker/src/MainFrame.cpp:595:13: error: ‘wxSAVE’ was not declared in this scope
   _T("*.h"),wxSAVE|wxOVERWRITE_PROMPT);
             ^~~~~~
/home/radek/slinker/src/MainFrame.cpp:595:13: note: suggested alternative: ‘wxFC_SAVE’
   _T("*.h"),wxSAVE|wxOVERWRITE_PROMPT);
             ^~~~~~
             wxFC_SAVE
/home/radek/slinker/src/MainFrame.cpp:595:20: error: ‘wxOVERWRITE_PROMPT’ was not declared in this scope
   _T("*.h"),wxSAVE|wxOVERWRITE_PROMPT);
                    ^~~~~~~~~~~~~~~~~~
/home/radek/slinker/src/MainFrame.cpp:595:20: note: suggested alternative: ‘wxFD_OVERWRITE_PROMPT’
   _T("*.h"),wxSAVE|wxOVERWRITE_PROMPT);
                    ^~~~~~~~~~~~~~~~~~
                    wxFD_OVERWRITE_PROMPT
/home/radek/slinker/src/MainFrame.cpp: In member function ‘void MainFrame::OnCheckRuleSetForRedundancy(wxCommandEvent&)’:
/home/radek/slinker/src/MainFrame.cpp:688:15: error: ‘wxOPEN’ was not declared in this scope
   _T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
               ^~~~~~
/home/radek/slinker/src/MainFrame.cpp:688:15: note: suggested alternative: ‘wxFC_OPEN’
   _T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
               ^~~~~~
               wxFC_OPEN
/home/radek/slinker/src/MainFrame.cpp:688:22: error: ‘wxFILE_MUST_EXIST’ was not declared in this scope
   _T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
                      ^~~~~~~~~~~~~~~~~
/home/radek/slinker/src/MainFrame.cpp:688:22: note: suggested alternative: ‘wxFD_FILE_MUST_EXIST’
   _T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
                      ^~~~~~~~~~~~~~~~~
                      wxFD_FILE_MUST_EXIST
/home/radek/slinker/src/MainFrame.cpp:712:16: error: ‘wxSAVE’ was not declared in this scope
    _T("*.txt"),wxSAVE|wxOVERWRITE_PROMPT);
                ^~~~~~
/home/radek/slinker/src/MainFrame.cpp:712:16: note: suggested alternative: ‘wxFC_SAVE’
    _T("*.txt"),wxSAVE|wxOVERWRITE_PROMPT);
                ^~~~~~
                wxFC_SAVE
/home/radek/slinker/src/MainFrame.cpp:712:23: error: ‘wxOVERWRITE_PROMPT’ was not declared in this scope
    _T("*.txt"),wxSAVE|wxOVERWRITE_PROMPT);
                       ^~~~~~~~~~~~~~~~~~
/home/radek/slinker/src/MainFrame.cpp:712:23: note: suggested alternative: ‘wxFD_OVERWRITE_PROMPT’
    _T("*.txt"),wxSAVE|wxOVERWRITE_PROMPT);
                       ^~~~~~~~~~~~~~~~~~
                       wxFD_OVERWRITE_PROMPT
```

`wx-config --version` returns `3.0.4`
`cmake --version` returns `3.13.1`
`gcc --version` returns `8.2.1`

 What could be wrong?

--- a/src/MainFrame.cpp
+++ b/src/MainFrame.cpp
@@ -285,7 +285,7 @@ void MainFrame::OnDemonstrateLoopGrowthRules(wxCommandEvent& event)
 bool MainFrame::AskUserForSolvingRulesFile()
 {
 	wxString filename = wxFileSelector(_T("Specify the solving rules file to use:"),_T(""),_T("*.txt"),_T("txt"),
-		_T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
+		_T("*.txt"),wxFD_OPEN|wxFD_FILE_MUST_EXIST);
 	if(filename.empty()) return false;
 	try {
 		this->solving_rules.clear();
@@ -586,13 +586,13 @@ void MainFrame::OnImportLoopyPuzzleString(wxCommandEvent& event)
 void MainFrame::OnConvertRulesFileToCCode(wxCommandEvent& event)
 {
 	wxString filename = wxFileSelector(_T("Specify the solving rules file to use:"),_T(""),_T("*.txt"),_T("txt"),
-		_T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
+		_T("*.txt"),wxFD_OPEN|wxFD_FILE_MUST_EXIST);
 	if(filename.empty()) return;
 	try {
 		vector<SlinkerGrid::TRule> rules;
 		SlinkerGrid::ReadRulesFromFile(string(filename.fn_str()),rules);
 		wxString output_filename = wxFileSelector(_T("Specify where to save the file:"),_T(""),_T("*.h"),_T("h"),
-		_T("*.h"),wxSAVE|wxOVERWRITE_PROMPT);
+		_T("*.h"),wxFD_SAVE|wxFD_OVERWRITE_PROMPT);
 		if(output_filename.empty()) return;
 		// write to file as a C function
 		ofstream out(output_filename.fn_str());
@@ -685,7 +685,7 @@ void MainFrame::OnShowTheSolution(wxCommandEvent &event)
 void MainFrame::OnCheckRuleSetForRedundancy(wxCommandEvent& event)
 {
 	wxString filename = wxFileSelector(_T("Specify the solving rules file to use:"),_T(""),_T("*.txt"),_T("txt"),
-		_T("*.txt"),wxOPEN|wxFILE_MUST_EXIST);
+		_T("*.txt"),wxFD_OPEN|wxFD_FILE_MUST_EXIST);
 	if(filename.empty()) return;
 	try 
 	{
@@ -709,7 +709,7 @@ void MainFrame::OnCheckRuleSetForRedundancy(wxCommandEvent& event)
 		else
 		{
 			wxString output_filename = wxFileSelector(_T("Redundant rules removed, specify where to save the new rules file:"),_T(""),_T("*.txt"),_T("txt"),
-			_T("*.txt"),wxSAVE|wxOVERWRITE_PROMPT);
+			_T("*.txt"),wxFD_SAVE|wxFD_OVERWRITE_PROMPT);
 			if(!output_filename.empty())
 				SlinkerGrid::WriteRulesToFile(nonredundant_rules,string(output_filename.fn_str()));
 		}

